llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -4966,7 +4966,7 @@ SDValue TargetLowering::SimplifySetCC(EVT VT, SDValue N0, SDValue N1,
         APInt C = C1 + 1;
         ISD::CondCode NewCC = (Cond == ISD::SETLE) ? ISD::SETLT : ISD::SETULT;
         if ((DCI.isBeforeLegalizeOps() ||
-             isCondCodeLegal(NewCC, VT.getSimpleVT())) &&
+             isCondCodeLegal(NewCC, OpVT.getSimpleVT())) &&
```

Comment:
This looks like an independent bug fix, which has a good chance of exposing issues or changing compiler output. 
You may consider submitting it as a separate patch.

Reply:
Sounds good, I've created https://github.com/llvm/llvm-project/pull/116082 to land this separately